File: scripts/projects-config.ts
import { resolve } from "node:path";

export type ManagedProject = {
  installCommand: string[];
  name: string;
  path: string;
  verifyCommands: string[][];
};

const home = process.env.HOME ?? "/home/sawyer";
const githubRoot = process.env.GITHUB_ROOT
  ? resolve(process.env.GITHUB_ROOT)
  : resolve(home, "github");

export const managedProjects: ManagedProject[] = [
  {
    installCommand: ["bun", "install"],
    name: "astro-web-template",
    path: resolve(githubRoot, "astro-web-template"),
    verifyCommands: [["bun", "run", "build"]],
  },
  {
    installCommand: ["bun", "install"],
    name: "astro-blog-template",
    path: resolve(githubRoot, "astro-blog-template"),
    verifyCommands: [
      ["bun", "run", "lint"],
      ["bun", "run", "typecheck"],
      ["bun", "run", "build"],
    ],
  },
];

File: scripts/projects_config.py
"""Managed project definitions for the scry-home CLI."""

from __future__ import annotations

from dataclasses import dataclass, field
from pathlib import Path

GITHUB = Path.home() / "github"


@dataclass
class ManagedProject:
    name: str
    path: Path
    install_command: list[str]
    verify_commands: list[list[str]] = field(default_factory=list)


MANAGED_PROJECTS: list[ManagedProject] = [
    ManagedProject(
        name="scry-home",
        path=GITHUB / "scry-home",
        install_command=["bun", "install"],
        verify_commands=[
            ["bun", "run", "lint"],
            ["uv", "run", "python", "-m", "scripts", "doctor"],
        ],
    ),
    ManagedProject(
        name="dunamismax",
        path=GITHUB / "dunamismax",
        install_command=["echo", "no install needed"],
        verify_commands=[],
    ),
    ManagedProject(
        name="boring-go-web",
        path=GITHUB / "boring-go-web",
        install_command=["go", "mod", "download"],
        verify_commands=[["go", "test", "./..."]],
    ),
    ManagedProject(
        name="c-from-the-ground-up",
        path=GITHUB / "c-from-the-ground-up",
        install_command=["echo", "no install needed"],
        verify_commands=[],
    ),
    ManagedProject(
        name="scryfall-discord-bot",
        path=GITHUB / "scryfall-discord-bot",
        install_command=["uv", "sync"],
        verify_commands=[
            ["uv", "run", "ruff", "check", "."],
            ["uv", "run", "mypy", "oracle"],
        ],
    ),
    ManagedProject(
        name="hello-world-from-hell",
        path=GITHUB / "hello-world-from-hell",
        install_command=["echo", "no install needed"],
        verify_commands=[["make", "test"]],
    ),
]
